### ISSUE TITLE
Cannot get SSL enabled.

### ISSUE THREAD

**Issue by sduensin** [saltcorn/saltcorn#1726]
I installed Saltcorn on Ubuntu 22.04 using npx as outlined on https://wiki.saltcorn.com/view/ShowPage/install-on-ubuntu .  I have enabled multitenancy.

I then provisioned a wildcard certificate per https://wiki.saltcorn.com/view/ShowPage/https-encryption and filled in my base URL.

Restarting Saltcorn, I still only see it listening on port 3000.  Running `saltcorn serve -v` shows it checking for a letsencrypt entry in the SQL but that's all I can see even relating to SSL.

Help!

**Comment by sduensin** [comment]
A dive into the code reveals this in `packages/server/serve.js`:
```
  // Server with http on port 80 / https on 443
  // todo  resolve hardcode
  if (port === 80 && cert && key) {
```
So Saltcorn only attempts to use the provided certificate and private key if the HTTP port is set to 80...

...but the installation instructions install Saltcorn under user `saltcorn` - who isn't allowed to use low ports!  Arrgh!

So, as root, I ran `setcap 'cap_net_bind_service=+ep' /usr/bin/node` to allow node.js to open any port it wants.  Not ideal, security-wise, but this box only runs Saltcorn.

&lt;Sigh&gt;  Still.  Saltcorn is awesome enough to deal with these sorts of issues!   

Hope this helps someone else.

**Comment by glutamate** [comment]
yes thats right. The setcap command should be run by `saltcorn-install` and already run on the digitalocean marketplace image.

Not sure what the best alternative here is - the thinking is that either you are running saltcorn as the non-proxied webserver and therefore you need certificates, or if you are running not on port 80 then you must have some proxy in front and that should be doing the SSL certificate. I realised I could be scenarios where this is not the case, so if you have ideas about some configuration options that could be added to deal with these possibilities I'd be happy to add them

**Comment by sduensin** [comment]
Something happened during my saltcorn-install and I had to finish it up manually.  Guess that's why it wasn't set for me.

After you explained the logic behind it, it makes sense.  I saw in the documentation on the wiki - later - about using port 80 with setcap.  Maybe it's more a documentation issue than anything.
